### ISSUE TITLE
alias grep='grep --color=auto' on alpine linux

### ISSUE THREAD

**Issue by ellerbrock** [Bash-it/bash-it#884]
hi there,

on alpine linux which is based on busybox we get an problem with the default alias for
`alias grep='grep --color=auto'`

when entering for example a .git folder:

```
± |master ✓| → l
.                .editorconfig    .travis.yml      aliases          custom           plugins          themes
..               .git             CONTRIBUTING.md  bash_it.sh       install.sh       template         uninstall.sh
.ackrc           .gitignore       README.md        completion       lib              test
grep: unrecognized option: color=auto
BusyBox v1.25.1 (2016-10-26 16:15:20 GMT) multi-call binary.

Usage: grep [-HhnlLoqvsriwFE] [-m N] [-A/B/C N] PATTERN/-e PATTERN.../-f FILE [FILE]...

Search for PATTERN in FILEs (or stdin)

	-H	Add 'filename:' prefix
	-h	Do not add 'filename:' prefix
	-n	Add 'line_no:' prefix
	-l	Show only names of files that match
	-L	Show only names of files that don't match
	-c	Show only count of matching lines
	-o	Show only the matching part of line
	-q	Quiet. Return 0 if PATTERN is found, 1 otherwise
	-v	Select non-matching lines
	-s	Suppress open and read errors
	-r	Recurse
	-i	Ignore case
	-w	Match whole words only
	-x	Match whole lines only
	-F	PATTERN is a literal (not regexp)
	-E	PATTERN is an extended regexp
	-m N	Match up to N times per file
	-A N	Print N lines of trailing context
	-B N	Print N lines of leading context
	-C N	Same as '-A N -B N'
	-e PTRN	Pattern to match
	-f FILE	Read pattern from file

 2017-01-09 11:22:02 ⌚  4a56f4ce9574 in ~/.bash_it
```

i would recommend to check first if this option is available, for example:

`[[ $(grep --color=auto 2>/dev/null) ]] && alias grep='grep --color=auto'`

i could not find it in `bash-it/aliases/available/general.aliases.bash` and bit short in time right now for further investigation. first want to ask if its ok to make that check for u and if u know where this is set?

cheers maik

**Comment by ellerbrock** [comment]
i add a quick fix for alpine linux in the docker repository by `unalias grep`.
we should think about porting this to the main repo or make a check like above mentioned

https://github.com/Bash-it/bash-it-docker/commit/c44b1b7f2d579c5f83410ce4105c021984acf921

**Comment by nwinkler** [comment]
This is similar to the following issues (for `ls`):

* #819 
* #643 

Ideally, we can fix both `ls` and `grep` in the same PR...

**Comment by nwinkler** [comment]
I'll take a stab at fixing these now...

**Comment by nwinkler** [comment]
Should be fixed now, see #885

**Comment by ellerbrock** [comment]
nice, thanks!
i updated the docker repo and removed the fix, also i added today new usefull features like running as unprivileged user and run dumb-init to prevent zombie processes (https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/)
